lib/src/client.dart

```diff
@@ -197,6 +213,11 @@ class ClientImpl implements Client {
     _token = token;
   }
 
+  @override
+  void setHeaders(Map<String, dynamic> headers) {
+    _headers = headers;
+  }
```

Comment:
A bit safer option to make copy of headers, because user can pass mutable HashMap and mutate it over time.

```dart
  @override
  void setHeaders(Map<String, Object?> headers) =>
    _headers = Map<String, Object?>.of(headers);
  ```
  or
  ```dart
  @override
  void setHeaders(Map<String, Object?> headers) => 
    _headers = <String, Object?>{
       for (final MapEntry(:String key, :Object? value) in headers.entries)
          if (value is String) key: value
          else if (value is Iterable) key: value.whereType<String>().join(', '),
    };
  ```

Reply:
Done